frappe/public/js/frappe/list/list_sidebar.js

```diff
@@ -224,39 +231,144 @@ frappe.views.ListSidebar = class ListSidebar {
 		});
 	}
 
-	setup_assigned_to() {
-		$('.assigned-loading').show();
-		let dropdown = this.page.sidebar.find('.assigned-dropdown');
-		let current_filters = this.list_view.get_filters_for_args();
-
-		frappe.call('frappe.desk.listview.get_user_assignments_and_count', {doctype: this.doctype, current_filters: current_filters}).then((data) => {
-			$('.assigned-loading').hide();
-			let current_user  = data.message.find(user => user.name === frappe.session.user);
-			if(current_user) {
-				let current_user_count = current_user.count;
-				this.get_html_for_assigned(frappe.session.user, current_user_count).appendTo(dropdown);
+	setup_list_group_by() {
```

Comment:
Lots of group by functionality is spread out in this file. Better to move all group by logic to another file, preferably a `class`